modules/post/osx/gather/enum_osx.rb

```diff
@@ -105,42 +105,19 @@ def log_folder_create(log_path = nil)
   # Checks if running as root on the target
   def check_root
     # Get only the account ID
-    case session.type
-    when /shell/
-      id = cmd_exec("/usr/bin/id -ru").chomp
-    when /meterpreter/
-      id = cmd_exec("/usr/bin/id", "-ru").chomp
-    end
-
-    if id == "0"
-      return true
-    else
-      return false
-    end
+    cmd_exec("/usr/bin/id", "-ru") == "0"
   end
 
   # Checks if the target is OSX Server
   def check_server
     # Get the OS Name
-    case session.type
-    when /meterpreter/
-      osx_ver = cmd_exec("/usr/bin/sw_vers", "-productName").chomp
-    when /shell/
-      osx_ver = cmd_exec("/usr/bin/sw_vers -productName").chomp
-    end
-    return osx_ver =~/Server/
+    cmd_exec("/usr/bin/sw_vers", "-productName") =~/Server/
```

Comment:
Regex is unnecessary here. `include?` would be cleaner.

`cmd_exec("/usr/bin/sw_vers", "-productName").to_s.include? 'Server'`